### PR TITLE
Rabbitmq/separate clients

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

This PR follows up https://github.com/ITISFoundation/osparc-simcore/pull/4682 where @pcrespov was suggesting separating the RabbitMQClient for standard RabbitMQ operations (pub/sub) and RPC operations https://github.com/ITISFoundation/osparc-simcore/pull/4682#discussion_r1309394428.
This PR implements just that and re-organizes a bit the different modules
After this PR:
- RabbitMQClient for standard pub/sub operations
- RabbitMQRPCClient for RPC operations

## Related issue/s
- relates to https://github.com/ITISFoundation/osparc-simcore/pull/4682

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: packages/service-library/src/servicelib/rabbitmq/_client_base.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                _logger.error(
+                    "Rabbit channel closed with exception from %s:%s",
+                    type(exc),
+                    exc,
+                )
+                self._healthy_state = False
+
+    @property
+    def healthy(self) -> bool:
+        return self._healthy_state
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/rabbitmq/_client_base.py:63]: Here healthy refers to the channel, not the rmq service right?

**Review by pcrespov — APPROVED** [review]
Fantastic! 🎉  Thanks for this addition.

File: packages/service-library/src/servicelib/rabbitmq/_client_base.py (hunk 1 above)
Comment (sanderegg) [reply packages/service-library/src/servicelib/rabbitmq/_client_base.py:63]: it refers to the client state. and is set to False when there are errors given when closing connection and/or channel

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc.py [hunk 2]
```diff
@@ -343,50 +320,22 @@ async def _requester_scope() -> None:
...
     with pytest.raises(RuntimeError) as exec_info:
-        await rabbit_replier.rpc_register_handler(
+        await rpc_server.register_handler(
             namespace, RPCMethodName("same_name"), _another_handler
         )
     assert "Method name already used for" in f"{exec_info.value}"
 
 
-async def test_rpc_register_for_is_equivalent_to_rpc_register(
-    rabbit_replier: RabbitMQClient,
```
Comment (GitHK) [inline packages/service-library/tests/rabbitmq/test_rabbitmq_rpc.py:365]: Q: why was this one removed?

**Review by GitHK — APPROVED** [review]

File: packages/service-library/tests/rabbitmq/test_rabbitmq_rpc.py (hunk 2 above)
Comment (sanderegg) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc.py:365]: Completely unused, and very confusing

Comment (sanderegg) [reply packages/service-library/tests/rabbitmq/test_rabbitmq_rpc.py:365]: @GitHK any use?

**Pull request merged**